server/src/utils.rs

```diff
@@ -79,6 +83,60 @@ where
 	}
 }
 
+/// Serve a service over a TCP connection.
+pub async fn serve<S, B, I>(io: I, service: S) -> Result<(), BoxError>
```

Comment:
nit: I would mention here that graceful shutdowns are not handled, and users should look at `serve_with_graceful_shutdown` in those cases or similar